### ISSUE TITLE
Regression of PZF compression

### ISSUE THREAD

**Issue by David-Baddeley** [python-microscopy/python-microscopy#80]
**[Original report](https://bitbucket.org/david_baddeley/python-microscopy/issue/80) by Andrew Barentine (Bitbucket: [AndrewBarentine](https://bitbucket.org/AndrewBarentine), ).**

----------------------------------------

Perhaps only on python2?

‌

```
ERROR:root:Exception whilst putting files
Traceback (most recent call last):
  File "/home/smeagol/code/python-microscopy/PYME/Acquire/HTTPSpooler.py", line 175, in _queuePoll
    pzf = PZFFormat.dumps(frame, sequenceID=self.sequenceID, frameNum = imNum, **self.compSettings)
  File "/home/smeagol/code/python-microscopy/PYME/IO/PZFFormat.py", line 274, in dumps
    dataString = bcl.HuffmanCompressQuant(d1, quantizationOffset, quantizationScale).tostring()
  File "work/bcl.pyx", line 26, in work.bcl.HuffmanCompressQuant (bcl.c:1983)
ValueError: Buffer has wrong number of dimensions (expected 1, got 3)
Exception in thread Thread-16:
Traceback (most recent call last):
  File "/home/smeagol/anaconda2/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/home/smeagol/anaconda2/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/smeagol/code/python-microscopy/PYME/Acquire/HTTPSpooler.py", line 175, in _queuePoll
    pzf = PZFFormat.dumps(frame, sequenceID=self.sequenceID, frameNum = imNum, **self.compSettings)
  File "/home/smeagol/code/python-microscopy/PYME/IO/PZFFormat.py", line 274, in dumps
    dataString = bcl.HuffmanCompressQuant(d1, quantizationOffset, quantizationScale).tostring()
  File "work/bcl.pyx", line 26, in work.bcl.HuffmanCompressQuant (bcl.c:1983)
ValueError: Buffer has wrong number of dimensions (expected 1, got 3)
```

**Comment by David-Baddeley** [comment]
**Original comment by David Baddeley (Bitbucket: [david_baddeley](https://bitbucket.org/david_baddeley), ).**

----------------------------------------

* Is this on a development build of PYMECompress, or the packages?
* What platform? I’m guessing linux, py 2.7?

I’m pretty sure that I’ve tested the current code on both python 2 and 3, although it’s possible that the pymecompress packages for 2.7 need rebuilding.

**Comment by David-Baddeley** [comment]
**Original comment by Andrew Barentine (Bitbucket: [AndrewBarentine](https://bitbucket.org/AndrewBarentine), ).**

----------------------------------------

This was on linux 2.7. I built the packages and it works fine on linux 3.6. 

I just looked back and saw that my py2.7 environment has two pymecompress entries. Cleared them both, reinstalled via conda \(now pymecompress              0.1.3                    py27\_1    david\_baddeley\) and get the same traceback as shown above. Also reran setup.py develop for python-microscopy in the py27 environment.

**Comment by David-Baddeley** [comment]
**Original comment by David Baddeley (Bitbucket: [david_baddeley](https://bitbucket.org/david_baddeley), ).**

----------------------------------------

This all points to the pymecompress package being out of date. What if you do a `setup.py develop` install for pymecompress \(rather than installing from the conda packages\)? I’m pretty sure that this is how I tested on OSX.

I’m almost certain that the long term solution for this is going to be: 

* increment pymecompress version 
* add pymecompres >= new version number to the pyme dependencies

**Comment by David-Baddeley** [comment]
**Original comment by Andrew Barentine (Bitbucket: [AndrewBarentine](https://bitbucket.org/AndrewBarentine), ).**

----------------------------------------

You’re right, @{david_baddeley} ! I’ve incremented and built conda packages, which now works on py27. Changes included in PR #247, and I’ll send the conda packages

**Comment by David-Baddeley** [comment]
**Original comment by Andrew Barentine (Bitbucket: [AndrewBarentine](https://bitbucket.org/AndrewBarentine), ).**

----------------------------------------

issue was in using new pyme with old pymecompress version
